sdk/resourcemanager/Azure.ResourceManager.Core/src/ApiVersions.cs

```diff
@@ -89,27 +89,48 @@ internal async Task<string> LoadApiVersionAsync(ProvidersOperations providers, R
                     return type.ApiVersions[0];
                 }
             }
-            return string.Empty;
+            return null;
+        }
+
+        internal string TyrGetApiVersion(ProvidersOperations providers, ResourceIdentifier resourceId, CancellationToken cancellationToken)
+        {
+            string val;
+            if (TryGetApiVersion(resourceId, out val))
+            {
+                return val;
+            }
+            return LoadApiVersion(providers, resourceId, cancellationToken);
+        }
+
+        internal async Task<string> TyrGetApiVersionAsync(ProvidersOperations providers, ResourceIdentifier resourceId, CancellationToken cancellationToken)
```

Comment:
not sure the dual tryget is the right approach here.

I think both internal and public we want

try extension
try manual override
try load from provider
return null / throw

right?